lib/cocoapods/installer/analyzer.rb

```diff
@@ -406,6 +407,39 @@ def generate_target(target_definition, pod_targets)
         target
       end
 
+      # Verify that targets using a pod have the same swift version
+      #
+      # @param  [Hash{Podfile::TargetDefinition => Array<Specification>}] specs_by_target
+      #         the resolved specifications grouped by target.
+      #
+      # @note raises Informative if targets using a pod do not have
+      #       the same swift version
+      #
+      def check_pod_target_swift_versions(specs_by_target)
+        targets_by_spec = {}
+        specs_by_target.each do |target, specs|
+          specs.each do |spec|
+            (targets_by_spec[spec] ||= []) << target
+          end
+        end
+
+        error_message_for_target = lambda do |target|
+          "#{target.name} (Swift #{target.swift_version})"
+        end
+
+        error_messages = targets_by_spec.map do |spec, targets|
+          swift_targets = targets.reject { |target| target.swift_version.blank? }
```

Comment:
Blank?  Checks for nil and empty.  I think there is actually a semantic difference between them in Xcode so I'm not so sure that we should normalise?